volmdlr/wires.py

```diff
@@ -2720,76 +2832,166 @@ def repositioned_polygon(self, x, y):
-                        if intr_list[0].point_distance(intr_list[1].start) < \
-                                intr_list[0].point_distance(intr_list[1].end):
-                            closing_point = intr_list[1].start
-                        else:
-                            closing_point = intr_list[1].end
-                elif intr_list[0] == middle_point and \
-                        primitive.length() == intr_list[1].length():
-                    closing_point = intr_list[1].start
+        return line_intersections
+
+    def select_farthest_sewing_closing_point(self,
+                                             line_segment: volmdlr.edges.LineSegment2D,
+                                             polygon_primitive,
+                                             possible_closing_points):
+        closing_point = volmdlr.O2D
+        middle_point = polygon_primitive.middle_point()
+        distance = 0
+        for intr_list in possible_closing_points:
+            if intr_list[1] not in self.line_segments:
+                dist = intr_list[0].point_distance(line_segment.start)
+                if dist > distance:
+                    distance = dist
+                    if intr_list[0].point_distance(intr_list[1].start) < \
+                            intr_list[0].point_distance(intr_list[1].end):
+                        closing_point = intr_list[1].start
+                    else:
+                        closing_point = intr_list[1].end
+            elif intr_list[0] == middle_point and \
+                    polygon_primitive.length() == intr_list[1].length():
+                closing_point = intr_list[1].start
+                distance = 0
+
+        return closing_point
+
+    def select_closest_sewing_closing_point(self,
+                                            line_segment: volmdlr.edges.LineSegment2D,
+                                            polygon_primitive,
+                                            possible_closing_points):
+        closing_point = volmdlr.O2D
+        middle_point = polygon_primitive.middle_point()
+        distance = math.inf
+        for intr_list in possible_closing_points:
+            if intr_list[1] not in self.line_segments:
+                dist = intr_list[0].point_distance(line_segment.start)
+                if dist < distance:
+                    distance = dist
+                    if intr_list[0].point_distance(intr_list[1].start) < \
+                            intr_list[0].point_distance(intr_list[1].end):
+                        closing_point = intr_list[1].start
+                    else:
+                        closing_point = intr_list[1].end
+            elif intr_list[0] == middle_point and \
+                    polygon_primitive.length() == intr_list[1].length():
+                closing_point = intr_list[1].start
+                distance = 0
+
+        return closing_point
+
+    def search_farthest(self, point, possible_closing_points):
```

Comment:
docstring

Reply:
It's been added in [this PR](https://github.com/Dessia-tech/volmdlr/pull/332)